File: Data Structures & Algorithms/majority-element/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        let count = 1;
        let el = nums[0];

        for(let i = 1; i < nums.length; i++){
            let num = nums[i];

            if(num === el){
                count++;
            } else{
                count--;
            }

            if(count === 0){
                el = num;
                count = 1;
            }
        }

        return el;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.cs
public class Solution {
    public List<List<int>> CombinationSum(int[] nums, int target) {
        var result = new List<List<int>>();
        var current = new List<int>();
        Array.Sort(nums);

        dfs(result, current, 0, target, nums);
        return result;
    }

    public void dfs(List<List<int>> result, List<int> current, int idx, int remainder, int[] nums){
        if(remainder == 0){
            result.Add(new List<int>(current));
            return;
        }

        for(int i = idx; i < nums.Length; i++){
            if(nums[i] > remainder) break;

            current.Add(nums[i]);
            dfs(result, current, i, remainder - nums[i], nums);
            current.RemoveAt(current.Count-1);
        }
    }
}
